source/opt/eliminate_dead_io_components_pass.cpp

```diff
@@ -236,7 +236,27 @@ void EliminateDeadIOComponentsPass::ChangeIOVarStructLength(Instruction& io_var,
     }
     type_mgr->AttachDecoration(*dec, &new_struct_ty);
   }
-  analysis::Type* reg_new_var_ty = type_mgr->GetRegisteredType(&new_struct_ty);
+  // Clone name instructions for new struct type
```

Comment:
this might be refactorized into a method to clone the struct copying names no?

Reply:
Done. Added to IRContext.